### PR TITLE
polarArray start angle, and rotation fix (#1006)

### LINKED ISSUE #1006: Workplane polarArray incorrect startAngle and angle (according to its docstring and convention)
From the `polarArray` docstring:
```
        :param startAngle: Starting angle (degrees) of array. 0 degrees is
            situated along local X-axis.
        :param angle: The angle (degrees) to fill with elements. A positive
            value will fill in the counter-clockwise direction.
```

Would you be opposed to correcting `polarArray` behavior to follow the docstring?

There are no doc examples and few other references of usage.  I think the issue wasn't noticed because usage was typically with `angle=360` e.g. #118.

### PR DESCRIPTION
**Pull request by lorenzncode**
To resolve #1006.

### CODE REVIEW

**Review by jmwright — APPROVED** [review]
Looks good, thanks @lorenzncode

File: cadquery/cq.py [hunk 1]
```diff
@@ -1486,57 +1486,47 @@ def polarArray(
...
-            loc = Location(Vector(x, y))
-
-        locs = [loc]
+        if count < 1:
+            raise ValueError(f"At least 1 element required, requested {count}")
 
         # Calculate angle between elements
         if fill:
-            if angle % 360 == 0:
+            if abs(math.remainder(angle, 360)) < 1e-6:
```
Comment (adam-urbanczyk) [inline cadquery/cq.py:1508]: Could you make this 1e-6 into a module constant, say `TOL`?

File: cadquery/cq.py [hunk 2]
```diff
@@ -1486,57 +1486,47 @@ def polarArray(
...
-            y = radius * math.cos(phi)
-
-            if rotate:
-                loc = Location(Vector(x, y), Vector(0, 0, 1), -phi_deg)
             else:
-                loc = Location(Vector(x, y))
+                # Inclusive start and end
+                angle = angle / (count - 1) if count > 1 else startAngle
+
+        def locs():
```
Comment (adam-urbanczyk) [inline cadquery/cq.py:1514]: I'd propose to keep the original linear flow here (i.e. no nested functions).

**Review by adam-urbanczyk — CHANGES_REQUESTED** [review]

File: cadquery/cq.py (hunk 1 above)
Comment (lorenzncode) [reply cadquery/cq.py]: Module constant added.  Perhaps in the future tolerance could be a constant at the package level as there is a similar comparison in `Sketch.parray`.  Then maybe a different constant is needed for angle comparison vs others.

File: cadquery/cq.py (hunk 2 above)
Comment (lorenzncode) [reply cadquery/cq.py]: Got it.  I thought I'd experiment with a generator but probably no benefit as `pushPoints` builds a list.

I'm having fun with CQ and certainly fortunate that I'm still able to do so.  Thanks for taking the time to review.

**Review by adam-urbanczyk — APPROVED** [review]

**Comment by adam-urbanczyk** [comment]
Thanks @lorenzncode !

**Pull request merged**